src/kernel/clock.c

```diff
@@ -35,3 +35,26 @@ void clock_adjust(timestamp wallclock_now, s64 temp_cal, timestamp sync_complete
     rtc_settimeofday(sec_from_timestamp(wallclock_now));
 }
 KLIB_EXPORT(clock_adjust);
+
+closure_function(0, 1, boolean, timer_id_rtc,
+                timer, t)
+{
+    switch (t->id) {
+    case CLOCK_ID_REALTIME:
+    case CLOCK_ID_REALTIME_COARSE:
+    case CLOCK_ID_REALTIME_ALARM:
+        return true;
+    default:
+        return false;
+    }
+}
+
+void clock_reset_rtc(timestamp wallclock_now)
+{
+    clock_debug("%s: now %T, wallclock_now %T\n",
+                __func__, now(CLOCK_ID_REALTIME), wallclock_now);
+    rtc_settimeofday(sec_from_timestamp(wallclock_now));
+    timer_adjust(runloop_timers, stack_closure(timer_id_rtc), wallclock_now - now(CLOCK_ID_REALTIME));
```

Comment:
It might be a little more accurate to call now prior to the rtc_settimeofday, as rtc_settimeofday will take up some time doing port I/O and re-reading the RTC (which could include spinning for a ready status). Sampling the clock after the RTC set will cause this additional time to be included in the delta as skew.

Reply:
Good point!